### DISCUSSION TITLE
generic TCP proxy

### DISCUSSION THREAD

**Discussion by Jackson-Pollock** [mitmproxy/mitmproxy#4988] [Q&A]
Can MITM decrypt non-browser traffic using generic TCP proxy? I tried but my .exe who is initiating TLS traffic with server that can't be observed in MITM. Sorry if my understanding is wrong.

**Comment by BeyondEvil** [comment]
I'm interested in this as well.

**Comment by mhils** [comment]
**In short: Yes.** See https://docs.mitmproxy.org/dev/concepts-protocols/#generic-tcp-proxy for details.

mitmproxy 7 automatically detects non-HTTP traffic and proxies it as pure TCP (the same applies for TCP over TLS). https://mitmproxy.org/posts/releases/mitmproxy7/ has an example of what that looks like in the UI. :)
